frontend/ui/trapper.lua

```diff
@@ -45,69 +52,74 @@ function Trapper:wrap(func)
--- @treturn bool true if code is wrapped by Trapper, false otherwise
+--
+-- @treturn boolean true if code is wrapped by Trapper, false otherwise
 function Trapper:isWrapped()
     if coroutine.running() then
         return true
     end
     return false
 end
 
---- Allow for overriding the Paused text and buttons on the abort
--- ConfirmBox displayed when InfoMessage dismissed in Trapper:UIFeedback()
-function Trapper:setPauseText(text, ok_text, cancel_text)
-    self.paused_text = text
-    self.paused_continue_text = ok_text
-    self.paused_abort_text = cancel_text
+--- Clear any left-over widget
+function Trapper:clear()
+    if self:isWrapped() then
+        if self.current_widget then
+            UIManager:close(self.current_widget)
+            UIManager:forceRePaint()
+            self.current_widget = nil
+        end
+    end
 end
 
---- Simple UI interaction via InfoMessage and ConfirmBox
---
--- Usage:
---
--- Trapper:UIFeedback("some text", true)
--- Trapper:UIFeedback("some text", true, "go on", "no more")
---   - display a ConfirmBox with the text and ok_text/cancel_text buttons.
---   - block and wait for user choice
---   - return the choice made: true if OK tapped, false if Cancel tapped or dismissed
---
--- Trapper:UIFeedback("some text")
--- Trapper:UIFeedback()
---   - display a InfoMessage with text
---     or keep existing InfoMessage if text = nil
---   - return true
---   UNLESS the previous widget was itself a InfoMessage and it has been
---   dismissed (by Tap on the screen), in which case:
---     - don't display the new InfoMessage
---     - return false
---   So, we can only know a InfoMessage has been dismissed when we
---   try to display a new one (we can't do better than that with coroutines).
---   So, don't hesitate to call it regularly (each call costs 100ms), between
---   steps of the work, to provide good responsiveness.
---   Trapper:UIFeedback() is a shortcut to get dismiss info while keeping
---   the existing InfoMessage displayed.
---
--- Trapper:UIFeedback(false)
---   Remove any remaining widget and consider the usage of Trapper done.
---
--- @text string text message, or nil/false
--- @confirmbox[opt=false] bool use ConfirmBox if true, otherwise use InfoMessage
--- @ok_text string text for ConfirmBox Ok button
--- @cancel_text string text for ConfirmBox Cancel button
--- @treturn bool true if ConfirmBox OK or InfoMessage not dismissed, false if Cancel or dismissed
-function Trapper:UIFeedback(text, show_confirmbox, ok_text, cancel_text)
-
-    -- We need to be called from an already created and running coroutine,
-    -- ie: Trapper:wrap() is executing the function calling us
+--- Clear any left-over widget and reset Trapper state
+function Trapper:reset()
+    self:clear()
+    -- Reset some properties
+    self.paused_text = nil
+    self.paused_continue_text = nil
+    self.paused_abort_text = nil
+    return true
+end
+
+--[[--
+Display an InfoMessage, and catch dismissal.
```

Comment:
Displays :-P And you can do some fun internal doc references here, or would you like me to do that?

Reply:
I won't take that fun from you :) So you can if you want.
(Dunno if you can add commit to my PR, or if it should be merged and you add after that)

Reply:
We'll just merge it as is and then I'll beautify it a little sometime later. I'm working on other things atm (only some of which are KOReader-related =P).

Reply:
OK.
You tell me if/when you're done with corrected my english :) and I'll rebase it into 2 commits (trapper&wikipedia + dict cancellable) for clarity.